pypeit/specobjs.py

```diff
@@ -433,19 +445,27 @@ def set_names(self):
 
     def add_sobj(self, sobj):
         """
-        Add one or more SpecObj
+        Append one or more SpecObj to the existing set.
 
         Args:
-            sobj (SpecObj or list or ndarray):  One or more SpecObj objects
-
+            sobj (:class:`~pypeit.specobj.SpecObj`, :class:`~pypeit.specobjs.SpecObjs`, :obj:`list`, `numpy.ndarray`_):
+                One or more SpecObj objects to append.  If a list or array, the
+                elements of these must be instances of
+                :class:`~pypeit.specobj.SpecObj`.
         """
         if isinstance(sobj, specobj.SpecObj):
             self.specobjs = np.append(self.specobjs, [sobj])
-        elif isinstance(sobj, (np.ndarray,list)):
-            self.specobjs = np.append(self.specobjs, sobj)
-        elif isinstance(sobj, SpecObjs):
+            return
+        if isinstance(sobj, SpecObjs):
+            # TODO: Is this loop necessary?  Why not np.append(self.specobjs, sobj.specobjs)?
```

Comment:
It might have been `sobj.specobjs` could be a list instead of an array.
But maybe that doesn't matter.

Reply:
Okay, I made a minor edit to the comment, but leaving it for now.